app/services/credit_notes/create_from_termination.rb

```diff
@@ -2,9 +2,10 @@
 
 module CreditNotes
   class CreateFromTermination < BaseService
-    def initialize(subscription:, reason: 'order_change')
+    def initialize(subscription:, reason: 'order_change', upgrade: false)
       @subscription = subscription
       @reason = reason
+      @upgrade = upgrade
```

Comment:
Is it possible to know that's an upgrade without sending it as an argument?

you already have the subscription, no?

Reply:
I first tried to do it, but all changes are performed inside the DB transaction and e.g. when I do a check to see if subscription is really updated, query does not return valid answer since new subscription at this point is not really stored.

Reply:
ok :)